content/chainguard/chainguard-images/videos/java-images.md

```diff
@@ -0,0 +1,173 @@
+---
+title: "How to Migrate a Java Application to Chainguard Images"
+linktitle: "Migrate Java Applications to Chainguard"
+lead: ""
+description: "How to migrate an existing Dockerfile for a Java application to use Chainguard Images
+in order to improve security and reduce image size."
+type: "article"
+date: 2024-04-02T15:21:01+00:00
+lastmod: 2024-04-02T15:21:01+00:00
+draft: false
+images: []
+menu:
+  docs:
+    parent: "chainguard-images"
+weight: 60
+toc: true
+---
+
+{{< youtube FYOVcSv1-oY >}}
+
+## Tools used in this video
+
+* [Docker](https://docker.com)
+
+## Resources
+
+* Blog on [bootstrapping Java in Wolfi](https://www.chainguard.dev/unchained/fully-bootstrapping-java-from-source-in-wolfi)
+* [Learning Labs Git repository](https://github.com/chainguard-dev/learning-labs-java) with code used in demo
+
+## Transcript
+
+Okay, I want to give a quick overview of how to use the Chainguard Java images.
+
+And in particular, I want to show how to port an existing application to use the Chainguard Java images.
+
+So our images are largely equivalent to the existing Java images that you can find on the Docker Hub, such as the Eclipse Temurin ones.
+
+The difference is that we're much more focused on producing minimal images with a low CVE count.
+
+We do build our own JDK and there's a really great blog on the Chainguard site that explains how we do this and how we bootstrap from the really early versions of Java, which I thoroughly recommend checking out and I will link in the notes.
+
+For this video I'm going to use an example created by my colleague Mauren Berti, so all the hard work in this video is actually down to her.
+
+So the starting point for this video is this example app.
+
+It's a Spring Boot app and all it does is listen on port 8080 and return "Hello world" effectively.
+
+So here is the Dockerfile and we can see it starts with "FROM maven".
+
+So this is using the Docker official image for maven which itself is built on top of Eclipse Temurin.
+
+All we're doing then is copying over some source code, building it with maven, doing a little bit of cleanup on this step to get rid of some build artifacts and then we're copying the jar file to the app directory and setting an entrypoint.
+
+So we should be able to build that fairly easily and now I should be able to run it.
+
+That was pretty quick because it was cached already.
+
+If you build it yourself it will take a little bit longer because it will have to download all the various dependencies.
+
+So let's see if I can get this right.
+
+docker run --rm -d --port 8080 to port 8080 on the host.
```

Comment:
I am having trouble escaping right for a suggestion, but what if you break it onto two lines, so you can have the command easy for someone to copy?

```
docker run --rm -d --port 8080 
```

to port 8080 on the host.